core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerServiceSpec.js

```diff
@@ -0,0 +1,402 @@
+            fallbacks: [],
+            default_outcome: null,
+            answer_groups: [],
+            confirmed_unclassified_answers: [],
+            customization_args: {
+              recommendedExplorationIds: []
+            },
+            id: 'EndExploration'
+          },
+          content: [
+            {
+              type: 'text',
+              value: ''
+            }
+          ],
+          unresolved_answers: {}
+        },
+        Introduction: {
+          param_changes: [],
+          classifier_model_id: null,
+          interaction: {
+            fallbacks: [],
+            default_outcome: {
+              dest: 'Introduction',
+              feedback: [
+                '<p>all others can be ruled out by definition of slope.</p>'
+              ],
+              param_changes: []
+            },
+            answer_groups: [
+              {
+                rule_specs: [
+                  {
+                    rule_type: 'Equals',
+                    inputs: {
+                      x: 0
+                    }
+                  }
+                ],
+                outcome: {
+                  dest: 'Question 1',
+                  feedback: [
+                    '<p>y/x is the measure of the slope.</p>'
+                  ],
+                  param_changes: []
+                }
+              }
+            ],
+            confirmed_unclassified_answers: [],
+            customization_args: {
+              choices: {
+                value: [
+                  '<p>y/x</p>',
+                  '<p>-y/x</p>',
+                  '<p>x/y</p>',
+                  '<p>-x/y</p>'
+                ]
+              }
+            },
+            id: 'MultipleChoiceInput'
+          },
+          content: [
+            {
+              type: 'text',
+              value: '<p><b>Linear algebra</b> is a subtopic of maths.</p>'
+            }
+          ],
+          unresolved_answers: {}
+        }
+      },
+      objective: ''
+    };
+
+    explorationStatesService.init(data.states);
+    explorationTitleService.init(data.title);
+    explorationInitStateNameService.init(data.init_state_name);
+
+    spyOn(validatorsService, 'isValidEntityName').andReturn(true);
+    spyOn(mockExplorationData, 'autosaveChangeList').andReturn(true);
+  });
+
+  it('should initialize the local data variables', function() {
+    var returnedValue = simpleEditorManagerService.tryToInit();
+    expect(simpleEditorManagerService.getTitle()).toBe(data.title);
+    expect(simpleEditorManagerService.getIntroductionHtml())
+      .toBe(data.states.Introduction.content[0].value);
+    var questions = statesToQuestionsService.getQuestions();
+    if (!questions) {
+      expect(returnedValue).toBeFalsy();
+    } else {
+      expect(returnedValue).toBeTruthy();
+    }
+    var questionList = questionListObjectFactory.create(questions);
+    expect(simpleEditorManagerService.getQuestionList()).toEqual(questionList);
```

Comment:
```javascript
var questionList = questionListObjectFactory.create(questions);
```
The question list here is transformed form of state data(that has been hard-coded earlier in beforeEach block, data.states => questionList).

I am generating the questionList instead of hardcoding it because the transformation logic of questionListObjectFactory.create() is not part of the unit under test and the questionListObjectFactory is assumed to be working corectly. hardcoding the questionList will require changes in hardcoded questionList if questionListObjectFactory.create() changes.